projects/plugins/jetpack/tests/e2e/lib/pages/homepage.js

```diff
@@ -0,0 +1,201 @@
+/**
+ * Internal dependencies
+ */
+import WpPage from './wp-page';
+import logger from '../logger';
+import { searchResultForTest1, searchResultForTest2 } from '../search-helper';
+
+export default class Homepage extends WpPage {
+	static SEARCH_API_PATTERN = /^https:\/\/public-api\.wordpress.com\/rest\/v1.3\/sites\/\d+\/search.*/;
+	static WAIT_FOR_ANIMATION_AND_RENDERING_TIMEOUT = 2000;
+
+	constructor( page ) {
+		const url = `${ siteUrl }/`;
+		super( page, { expectedSelectors: [ '.site-title' ], url } );
+	}
+
+	async registerRouteInterceptions() {
+		await this.searchAPIRoute();
+	}
+
+	searchAPIRoute() {
+		return this.page.route( Homepage.SEARCH_API_PATTERN, ( route, request ) => {
+			logger.info( `intercepted search API call: ${ request.url() }` );
+			const url = new URL( request.url() );
+			const params = url.searchParams;
+
+			// load response for queries
+			let body;
+			switch ( params.get( 'query' ) ) {
+				case 'test1':
+					body = { ...searchResultForTest1 };
+					break;
+				case 'test2':
+				default:
+					body = { ...searchResultForTest2 };
+					break;
+			}
+
+			// sorting
+			switch ( params.get( 'sort' ) ) {
+				case 'date_asc':
+					// put record 2 first
+					const tmpResult1 = body.results[ 0 ];
+					body.results[ 0 ] = body.results[ 1 ];
+					body.results[ 1 ] = tmpResult1;
+					break;
+				case 'date_desc':
+					// put record 3 first
+					const tmpResult2 = body.results[ 0 ];
+					body.results[ 0 ] = body.results[ 2 ];
+					body.results[ 2 ] = tmpResult2;
+					break;
+				case 'score_default':
+				default:
+					// the original sorting
+					break;
+			}
+
+			// filtering
+			// filter[bool][must][0][term][category.slug]=category-1
+			// filter[bool][must][2][term][tag.slug]=tag-1
```

Comment:
Is this comment still needed? If so, can you elaborate on what it means?

Reply:
removed in https://github.com/Automattic/jetpack/pull/19447/commits/9a282cad876e146369da6936770672802c493bb6